hanjpkeyboard.c

```diff
@@ -5,7 +5,7 @@ G_DEFINE_INTERFACE(HanjpKeyboard, hanjp_keyboard, G_TYPE_OBJECT)
 
 static void
 hanjp_keyboard_default_init(HanjpKeyboardInterface *iface) {
-    /* add properties and signals to the interface here */
+	iface->get_mapping = hanjp_keyboard_get_mapping;
```

Comment:
Connecting the member functions with your implementation must be written in hanjp_keyboarddefault_keyboard_interface_init function, not here, and the function is already written as far as I remember.

Reply:
yeah it was implemented under `hanjp_keyboarddefault_init` which can be found on the bottom of the code. But as you can see from the comment - that I did not wrote - it was their to be implemented so I did it anyway.

Reply:
No, it's a function for signals and properties, not for initialization.
I fix it.